django/db/models/functions/json.py

```diff
@@ -5,6 +5,69 @@
 from django.db.models.functions import Cast
 
 
+class JSONArray(Func):
+    function = "JSON_ARRAY"
+    output_field = JSONField()
+
+    def as_sql(self, compiler, connection, **extra_context):
+        if not connection.features.has_json_object_function:
+            raise NotSupportedError(
+                "JSONArray() is not supported on this database backend."
+            )
+        return super().as_sql(compiler, connection, **extra_context)
+
+    def as_native(self, compiler, connection, *, returning, **extra_context):
+        """Modify JSON_ARRAY for Oracle and Postgres 16 and later.
+
+        Postgres and Oracle both remove SQL NULL values from the array by
+        default. Adds the NULL ON NULL clause to keep NULL values in the array,
+        mapping them to JSON null values, which matches the behavior of SQLite.
+        """
```

Comment:
I'd chop the first sentence. Also, `Postgres` -> `PostgreSQL`:
```suggestion
        """
        PostgreSQL 16+ and Oracle remove SQL NULL values from the array by
        default. Adds the NULL ON NULL clause to keep NULL values in the array,
        mapping them to JSON null values, which matches the behavior of SQLite.
        """
```